### PR TITLE
Add Gulool Ja Ja

### PR DESCRIPTION
**Pull request by dwn134**
scripted his 1 hour added mobs to spawn on engage respawn 10 seconds if killed, if a holder dies then the adds will then agro to the tank. Adds despawn of boss goes to disengage despawns or dies

### CODE REVIEW

**Comment by whasf** [comment]
Look like you have a problem:

Importing mob_skill_lists.sql into the database...ERROR 1062 (23000) at line 1237: Duplicate entry '285-1801' for key 'PRIMARY'

File: scripts/zones/Mamook/MobIDs.lua [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-----------------------------------
+-- Area: Mamook
+-----------------------------------
+
+Gulool_Ja_Ja = 17043875;
```
Comment (wrenffxi) [inline scripts/zones/Mamook/MobIDs.lua]: For consistency, capitalize entries in MobIDs.  GULOOL_JA_JA

File: scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua [hunk 2]
```diff
@@ -4,31 +4,63 @@
...
 require("scripts/globals/titles");
-
------------------------------------
--- onMobSpawn Action
------------------------------------
+require("scripts/globals/status");
+require("scripts/zones/Mamook/MobIDs");
 
 function onMobSpawn(mob)
+    mob:setLocalVar("1Hour", math.random(3,5));
```
Comment (wrenffxi) [inline scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua]: there's now a mixin you can use to give a mob its 1hr/job special, so this line and the code block in onMobFight aren't needed.  See
https://github.com/DarkstarProject/darkstar/blob/e46bce5064fa0631c4a17b34837bea4ecb50aa81/scripts/zones/Lower_Delkfutts_Tower/mobs/Epialtes.lua#L5

File: scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua [hunk 3]
```diff
@@ -4,31 +4,63 @@
...
------------------------------------
--- onMobSpawn Action
------------------------------------
+require("scripts/globals/status");
+require("scripts/zones/Mamook/MobIDs");
 
 function onMobSpawn(mob)
+    mob:setLocalVar("1Hour", math.random(3,5)); 
+    mob:setMod(MOD_DOUBLE_ATTACK, 20);
 end;
```
Comment (wrenffxi) [inline scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua]: Also uses draw-in, which can be set as mobmod.  second param is 1 for single player, or 2 for alliance.  See:
https://github.com/DarkstarProject/darkstar/blob/ac0912ae626aace31fdaf0a41726a64af9570015/scripts/zones/The_Boyahda_Tree/mobs/Voluptuous_Vivian.lua#L10

File: scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua [hunk 4]
```diff
@@ -4,31 +4,63 @@
...
 
------------------------------------
--- onMobEngaged Action
------------------------------------
-
 function onMobEngaged(mob,target)
+    for i = 17043876, 17043879, 1 do
```
Comment (wrenffxi) [inline scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua]: entity IDs should be in MobIDs.lua.  recommend to simply do
```for i = GULOOL_JA_JA + 1, GULOOL_JA_JA + 4 do```

File: scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua [hunk 5]
```diff
@@ -4,31 +4,63 @@
...
+        SpawnMob(i):updateEnmity(target);
+    end
+end;
 
-    SpawnMob(17043876):updateEnmity(target);
-    SpawnMob(17043877):updateEnmity(target);
-    SpawnMob(17043878):updateEnmity(target);
-    SpawnMob(17043879):updateEnmity(target);
+function onMobFight(mob, target)
+    local petIDs = {17043876,17043877,17043878,17043879};
```
Comment (wrenffxi) [inline scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua]: same here. don't hard-code ids in entity luas (mob, npc, or text ids)

File: scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua [hunk 6]
```diff
@@ -4,31 +4,63 @@
...
+    local petStatus = {GetMobAction(petIDs[1]),GetMobAction(petIDs[2]),GetMobAction(petIDs[3]),GetMobAction(petIDs[4])};
```
Comment (wrenffxi) [inline scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua]: we're phasing out global function GetMobAction.  Instead, use ... {cont'd below}

File: scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua [hunk 7]
```diff
@@ -4,31 +4,63 @@
...
 
+    if (mob:getLocalVar("use1Hour") == 0) then
+        if (HPP <= mob:getLocalVar("1Hour")) then
+            mob:useMobAbility(731); -- Mijin Gakure
+            mob:setLocalVar("use1Hour", 1);
+        end
+    end
+
+    if (mob:getBattleTime() % 15 < 2 and mob:getBattleTime() > 10) then
+        if (GetMobAction(Gulool_Ja_Ja+1) == 0) then
```
Comment (wrenffxi) [inline scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua]: {cont'd}

```if (GetMobByID(foo):isDead()) then```

File: scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua [hunk 8]
```diff
@@ -4,31 +4,63 @@
...
+        elseif (GetMobAction(Gulool_Ja_Ja+3) == 0) then
+            GetMobByID(Gulool_Ja_Ja+3):setSpawn(mob:getXPos()+math.random(1,5), mob:getYPos(), mob:getZPos()+math.random(1,5));
+            SpawnMob(Gulool_Ja_Ja+3):updateEnmity(target);
+        elseif (GetMobAction(Gulool_Ja_Ja+4) == 0) then
+            GetMobByID(Gulool_Ja_Ja+4):setSpawn(mob:getXPos()+math.random(1,5), mob:getYPos(), mob:getZPos()+math.random(1,5));
+            SpawnMob(Gulool_Ja_Ja+4):updateEnmity(target);
+        end
+    end
+    for i = 1, 4 do
+        if petStatus[i] == 16 then
```
Comment (wrenffxi) [inline scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua]: {cont'd} or
```if (GetMobByID(foo):getCurrentAction() == ACTION_ROAMING) then```

File: scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua [hunk 9]
```diff
@@ -6,61 +6,54 @@
...
-    local HPP = mob:getHPP();
-
-    if (mob:getLocalVar("use1Hour") == 0) then
-        if (HPP <= mob:getLocalVar("1Hour")) then
-            mob:useMobAbility(731); -- Mijin Gakure
-            mob:setLocalVar("use1Hour", 1);
-        end
-    end
+    local petIDs = {GULOOL_JA_JA + 1, GULOOL_JA_JA + 2, GULOOL_JA_JA + 3, GULOOL_JA_JA + 4};
+    local petStatus = {GetCurrentAction(petIDs[1]),GetCurrentAction(petIDs[2]),GetCurrentAction(petIDs[3]),GetCurrentAction(petIDs[4])};
```
Comment (wrenffxi) [inline scripts/zones/Mamook/mobs/Gulool_Ja_Ja.lua]: almost right.  getCurrentAction is a lua_baseentity binding.  entire onMobFight could be simply:
https://pastebin.com/raw/dtmqvQm7

**Comment by TeoTwawki** [comment]
rebase to fix conflict pls, looks like its all good aside from that

**Pull request merged**